### DISCUSSION TITLE
Pagination links being removed when transforming DataCollection

### DISCUSSION THREAD

**Discussion by bmw27** [spatie/laravel-data#98] [Q&A] [answered]
I was messing around with this package and noticed it removes the links collection provided by Laravel's `LengthAwarePaginator` when transforming a DataCollection.

https://github.com/spatie/laravel-data/blob/b1c8975c72918078f6212f88dc3e7a2e69b0378d/src/Transformers/DataCollectionTransformer.php#L68-L77

Is there a way to override this behavior? I'm new to using data objects and API resources in general, so sorry if this is a dumb question 😅.

**Comment by rubenvanassche** [comment, marked as answer]
Hi @bmw27,

Good point! This should be fixed in the release later today!

**Reply by bmw27** [reply]
Awesome! Thanks for all the work you and Spatie do.

**Comment by didac-adria** [comment]
Hey!

I would like to re-open this topic. I have noticed that the links collection is organized differently than how Laravel does it currently. 

Laravel returns Links full info inside metadata and a short version at the same level of metadata and data:
![image](https://user-images.githubusercontent.com/63950600/179781032-92c542b1-09a6-4da9-b41c-ccfe0d11ef0d.png)

But Laravel-data package returns only full version at the same level of metadata and data:
![image](https://user-images.githubusercontent.com/63950600/179781418-44923255-15c0-421b-8244-a15d2d6bab3a.png)

This is a problem because our pagination logic on the frontend is re-used over all endpoints. In most of them, we use pagination as per Laravel's structure. But if we return a collection with the laravel-data package, we need to implement different logic for pagination in the frontend.

I think it's interesting to use the same structure as laravel for pagination data. What do you think @rubenvanassche ?

Thanks in advance!

**Reply by rubenvanassche** [reply]
Yeah you're right, you can always PR such feature!

**Reply by didac-adria** [reply]
yeah, we will have to investigate exactly how this works. We will have a look if we can. Thanks!
@albert-mestres

**Reply by davidkvasnovsky** [reply]
Has this already been implemented, or are you open to PR? @rubenvanassche @didac-adria 

I think it would be nice if we could unify pagination behavior between native Laravel and the Laravel-data package.

**Reply by didac-adria** [reply]
Hey!
I didn't find the time to work on this. Unless it has been implemented in any package updates since then, I guess they will welcome your PR.
